aiida/engine/daemon/client.py

```diff
@@ -95,6 +102,14 @@ def daemon_name(self) -> str:
         """
         return self._DAEMON_NAME.format(name=self.profile.name)
 
+    @contextmanager
+    def lock(self):
+        if self.is_daemon_running:
+            raise RuntimeError(f"Cannot lock the '{self.profile.name}' daemon whilst it is running.")
+        Path(self.daemon_lock_file).touch()
+        yield self
+        Path(self.daemon_lock_file).unlink()
```

Comment:
I think you actually want to do a:
```python
try:
   ...
   yield self
finally:
   Path(self.daemon_lock_file).unlink()
```
to ensure the file is removed even if there are exceptions

Reply:
Oh yeh, I was mistaking it with pytest fixtures, that have this try/finally logic built-in